plot-builder-portable/src/commonMain/kotlin/jetbrains/datalore/plot/builder/assemble/LegendAssembler.kt

```diff
@@ -163,6 +163,28 @@ class LegendAssembler(
 
     companion object {
         private const val DEBUG_DRAWING = jetbrains.datalore.plot.FeatureSwitch.LEGEND_DEBUG_DRAWING
+        fun wrap(text: String): String {
+            if (text.contains("\n") || text.length <= Legend.LINES_MAX_LENGTH) return text
+
+            return generateSequence(text.split(Legend.LINES_MAX_LENGTH)) {
+                when {
+                    it.first.isEmpty() -> null
+                    else -> it.first.split(Legend.LINES_MAX_LENGTH)
+                }
+            }
+                .map(Pair<*, String>::second)
+                .joinToString("\n", limit = Legend.LINES_MAX_COUNT)
+        }
+
+        private fun String.split(n: Int): Pair<String, String> {
+            return Pair(this.drop(this.getLineLength(n)), this.take(this.getLineLength(n)))
+        }
+
+        private fun String.getLineLength(n: Int): Int {
+            if (this.length <= n) return this.length
+            //Trying to find space in last part of string. If word length is greater 2 * n / 3, then we can split it
+            return if (this.take(n).lastIndexOf(" ") < n / 3) n else this.take(n).lastIndexOf(" ")
```

Comment:
Could you add unit tests, please? This logic doesn't look obvious, so this code can break after a few refactoring cycles.